src/Dialog/Dialog.tsx

```diff
@@ -366,15 +366,10 @@ const Footer = styled.div<SxProp>`
   ${sx};
 `
 
-const buttonTypes = {
-  normal: Button,
-  primary: ButtonPrimary,
-  danger: ButtonDanger,
-}
 const Buttons: React.FC<React.PropsWithChildren<{buttons: DialogButtonProps[]}>> = ({buttons}) => {
   const autoFocusRef = useProvidedRefOrCreate<HTMLButtonElement>(buttons.find(button => button.autoFocus)?.ref)
-  let autoFocusCount = 0
-  const [hasRendered, setHasRendered] = useState(0)
+  let autoFocusCount = -1
+  const [hasRendered, setHasRendered] = useState(-1)
```

Comment:
I think we should take this back. This manages auto focus when there is a footer button that has `autoFocus: true` prop.

[Story reference](https://github.com/primer/react/blob/main/src/stories/Dialog.stories.tsx#L129), [docs reference](https://primer.style/react/drafts/Dialog#keyboard)

Reply:
I did this change mostly because I saw the following:
- opened the dialog
- pressed `TAB` once
- no button gets focus
- pressed `TAB` one more time
- focus is on the 2nd button

Felt a bit weird, don't know if it's intended but also seems out of scope of this PR so I'll rollback this change and can get tackled in a new issue, if that's the case

Reply:
Hmm, interesting - I don't see this behaviour. Here is my local. I might be missing something but as you mentioned it is out of this Pr's scope so thank you for taking this back. 

https://user-images.githubusercontent.com/1446503/233222234-b03d6764-c020-452d-a2f1-6a5e7b642b82.mp4

_Video description: Dialog storybook page is on the screen and there is a button called "Open Dialog". User tabs into the button and opens the dialog and keep pressing the tab to move the trapped focus on the dialog._